utils/build-script-impl

```diff
@@ -2217,6 +2282,9 @@ for host in "${ALL_HOSTS[@]}"; do
                     -DENABLE_TESTING=YES
                     -DBUILD_SHARED_LIBS=$([[ ${product} == libdispatch_static ]] && echo "NO" || echo "YES")
                   )
+                  if [[ $(is_cross_tools_host ${host}) ]] ; then
+                      add_cross_cmake_options ${host} cmake_options
```

Comment:
This is the first time that we are using CMake cross compilation under the hood in build-script-impl. I wonder if it would make sense to always cross compile everything in build-script-impl. I don't know how much work that would entail, but it is work that would need to be done. Especially since @shahmishal is going to need to fix the cross compilation loop for build-script products.

Reply:
What do you mean by "cross-compile everything," explicitly pass in all these compiler and sysroot flags when natively compiling too? That might make sense, but as you said, would be a fair amount of work to track them all down. Of course, the benefit is that every natively supported platform becomes easily cross-compilable from other platforms.

As for fixing "the cross compilation loop for build-script products," I have a pull forthcoming with the tweaks I had to make to build-script when cross-compiling SPM, as that's a build-script product not built here in build-script-impl.